src/storage/connectionStorage.ts

```diff
@@ -166,7 +180,40 @@ export class ConnectionStorage {
     );
   }
 
-  async loadConnections() {
+  _loadPresetSavedConnections(): LoadedConnection[] {
+    const configuration = vscode.workspace.getConfiguration('mdb');
+    const presetSavedConnectionsInfo = configuration.inspect<
+      PresetSavedConnection[]
+    >('presetSavedConnections');
+
+    if (!presetSavedConnectionsInfo) {
+      return [];
+    }
+
+    const combinedPresetSavedConnections: PresetSavedConnectionWithSource[] = [
+      ...(presetSavedConnectionsInfo?.globalValue ?? []).map((preset) => ({
+        ...preset,
+        source: 'globalSettings' as const,
+      })),
+      ...(presetSavedConnectionsInfo?.workspaceValue ?? []).map((preset) => ({
+        ...preset,
+        source: 'workspaceSettings' as const,
+      })),
+    ];
+
+    return combinedPresetSavedConnections.map((presetConnection) => ({
+      id: uuidv4(),
```

Comment:
Why do we generate different ids here as opposed to using the name? Is it because we want to allow multiple connections with the same name as presets?

Reply:
That may be a good enough reason, It is also because storage saved connections use UUIDs and we end up combining the two, so was kind of thinking of it as a stricter way to stick to a "type" where IDs are arbitrary and not tied to the connection's contents.